translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,294 @@
+
+
+
+# function to get event status
+def get_event_status(start_time_string, end_time_string):  
+
+    start_time = datetime.fromtimestamp(start_time_string)
+    current_time = datetime.now()
+    future_date_after_2weeks = current_time + \
+                        timedelta(days = 14)
+    event_status = "active"
+    if current_time < start_time:
+        if start_time < future_date_after_2weeks  :
+            event_status = "pending"
+        else:
+            event_status = "planned"
+
+
+    elif end_time_string:
+        end_time = datetime.fromtimestamp(end_time_string)
+        if end_time < current_time:
+            event_status = "completed"
+    return event_status
+
+
+
+# Parse COtrip alert to WZDx
+def parse_alert(alert, callback_function=None):
+    if not validate_alert(alert):
+        if callback_function:    
+            callback_function(alert)
+        return None
+    event = alert.get('event', {})
+    feature = {}
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = parse_polyline(event['geometry'])
+
+    feature['type'] = "Feature"
+    properties = {}
+
+    # road_event_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['road_event_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # data_source_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['data_source_id'] = ''
+
+    # start_date
+    properties['start_date'] = reformat_datetime(event['header']['start_timestamp'])
+
+    # end_date
+    properties['end_date'] = reformat_datetime(event['header'].get('end_timestamp'))
+
+    # start_date_accuracy
+    properties['start_date_accuracy'] = "estimated"
+
+    # end_date_accuracy
+    properties['end_date_accuracy'] = "estimated"
+
+    # beginning_accuracy
+    properties['beginning_accuracy'] = "estimated"
+
+    # ending_accuracy
+    properties['ending_accuracy'] = "estimated"
+
+    # road_name
+    properties['road_name'] = event['detail']['road_name']
+
+    # direction
+    Direction_map = {'North': 'northbound', 'South': 'southbound', 'West': 'westbound', 'East': 'eastbound'}
+ 
+    properties['direction'] = Direction_map.get(event['detail']['direction'])
+
+    # vehicle impact
+    properties['vehicle_impact'] = 'unknown'
+
+    # Relationship
+    properties['relationship'] = {}
+
+    # lanes
+    properties['lanes'] = []
+
+    # beginning_cross_street
+    properties['beginning_cross_street'] = ""
+
+    # beginning_cross_street
+    properties['ending_cross_street'] = ""
+
+    # event status
+    properties['event_status'] = get_event_status(event['header']['start_timestamp'], event['header'].get('end_timestamp'))
+
+    # type_of_work
+    # maintenance, minor-road-defect-repair, roadside-work, overhead-work, below-road-work, barrier-work, surface-work, painting, roadway-relocation, roadway-creation
+    properties['types_of_work'] = []
```

Comment:
Are we not adding `types_of_work` or `restrictions`? Using `cotrip_1.json` as an example, I see "roadway-relocation" and "Flagger Operations" but they are excluded from the output file.

Reply:
@abinashkonersman If we've got those values in the input file, we'll want them as part of our output.

Reply:
should be fixed